Common/Brokerages/InteractiveBrokersBrokerageModel.cs

```diff
@@ -62,24 +62,38 @@ public override bool CanSubmitOrder(Security security, Order order, out Brokerag
         {
             message = null;
 
+            // validate security type
+            if (security.Type != SecurityType.Equity &&
+                security.Type != SecurityType.Forex &&
+                security.Type != SecurityType.Option &&
+                security.Type != SecurityType.Future)
+            {
+                message = new BrokerageMessageEvent(BrokerageMessageType.Warning, "NotSupported",
+                    "This model does not support " + security.Type + " security type."
+                );
+
+                return false;
+            }
+
+            // validate order quantity
             //https://www.interactivebrokers.com/en/?f=%2Fen%2Ftrading%2FforexOrderSize.php
-            switch (order.SecurityType)
+            if (security.Type == SecurityType.Forex &&
+                !IsForexWithinOrderSizeLimits(order.Symbol.Value, order.Quantity, out message))
+            {
+                return false;
+            }
+
+            // validate time in force
+            if (order.TimeInForce != TimeInForce.GoodTilCancelled && order.TimeInForce != TimeInForce.Day)
```

Comment:
Similar comment to the above, often cleaner to define a collection of supported cases and check the collection. This makes the code read nicer and makes it easier to update the conditions in the future as we add/remove support for various things.